### PR TITLE
Backend/feat: #87 adding drivers currently on ride in search pool

### PR DESCRIPTION
**Pull request by 2019UCP1350**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [x] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?
using postman

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/Merchant/DriverPoolConfig.hs [hunk 1]
```diff
@@ -50,6 +50,9 @@ mkPersist
       poolSortingType PoolSortingType
       singleBatchProcessTime Seconds
       tripDistance Meters
+      radiusShrinkValueForDriversOnRide Int
+      driverToDestinationDistanceThreshold Meters
+      driverToDestinationDuration Seconds
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Tabular/Merchant/DriverPoolConfig.hs:55]: is it also threshold?

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs [hunk 2]
```diff
@@ -554,6 +558,107 @@ getNearestDrivers mbVariant LatLong {..} radiusMeters merchantId onlyNotOnRide m
...
+    where_ $
+      personInfo ^. PersonRole ==. val Person.DRIVER
+        &&. personInfo ^. PersonMerchantId ==. val (toKey merchantId)
+        &&. driverInfo ^. DriverInformationActive
+        &&. driverInfo ^. DriverInformationOnRide
+        &&. not_ (driverInfo ^. DriverInformationBlocked)
+        &&. ( val (Mb.isNothing mbDriverPositionInfoExpiry)
+                ||. (locationInfo ^. DriverLocationCoordinatesCalculatedAt +. Esq.interval [Esq.SECOND $ maybe 0 getSeconds mbDriverPositionInfoExpiry] >=. val now)
+            )
+        &&. whenJust_ mbVariant (\var -> vehicleInfo ^. VehicleVariant ==. val var)
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs:641]: change this logic according to downgrade option added recently

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs [hunk 3]
```diff
@@ -150,6 +154,16 @@ prepareDriverPoolBatch driverPoolCfg searchReq batchNum = withLogTag ("BatchNum-
       let pickupLoc = searchReq.fromLocation
       let pickupLatLong = LatLong pickupLoc.lat pickupLoc.lon
       calculateDriverPoolWithActualDist DriverSelection driverPoolCfg (Just vehicleVariant) pickupLatLong merchantId True (Just radiusStep)
+    calcDriverCurrenlyOnRidePool radiusStep reduceRadiousValue = do
+      let merchantId = searchReq.providerId
+      transporter <- CTC.findByMerchantId merchantId >>= fromMaybeM (TransporterConfigDoesNotExist merchantId.getId)
+      if transporter.includeDriverCurreltOnRide && (radiusStep - 1) > 0
```
Comment (umangj-28) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs:160]: typo `includeDriverCurreltOnRide`

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Merchant/DriverPoolConfig.hs [hunk 4]
```diff
@@ -39,7 +39,10 @@ data DriverPoolConfigD u = DriverPoolConfig
     singleBatchProcessTime :: Seconds,
     tripDistance :: Meters,
     createdAt :: UTCTime,
-    updatedAt :: UTCTime
+    updatedAt :: UTCTime,
+    radiusShrinkValueForDriversOnRide :: Int,
+    driverToDestinationDistanceThreshold :: Meters,
+    driverToDestinationDuration :: Seconds
   }
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/Merchant/DriverPoolConfig.hs:46]: can you leave `createdAt` and `updatedAt` as last fields, for better readability?

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs [hunk 5]
```diff
@@ -530,3 +665,6 @@ computeActualDistance orgId pickup driverPoolResults = do
           intelligentScores = IntelligentScores Nothing Nothing Nothing Nothing defaultPopupDelay,
           isPartOfIntelligentPool = False
         }
+
+chanegeIntoDriverPoolResult :: DriverPoolResultCurrentlyOnRide -> DriverPoolResult
+chanegeIntoDriverPoolResult DriverPoolResultCurrentlyOnRide {..} = DriverPoolResult {..}
```
Comment (umangj-28) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs:670]: typo `chanege`

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs [hunk 6]
```diff
@@ -16,6 +16,9 @@
 module SharedLogic.DriverPool
   ( calculateDriverPool,
     calculateDriverPoolWithActualDist,
+    calculateDriverCurrentlyOnRideWithActualDist,
+    calculateDriverPoolCurrentlyOnRide,
+    chanegeIntoDriverPoolResult,
```
Comment (umangj-28) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs:21]: please resolve typos `chanege`

**Review by hkmangla — APPROVED** [review]

**Pull request merged**